types/wlr_cursor.c

```diff
@@ -472,10 +513,50 @@ static void handle_layout_destroy(struct wl_listener *listener, void *data) {
 	wlr_cursor_detach_output_layout(state->cursor);
 }
 
+static void handle_layout_output_destroy(struct wl_listener *listener,
+		void *data) {
+	struct wlr_cursor_output_cursor *output_cursor =
+		wl_container_of(listener, output_cursor, layout_output_destroy);
+	//struct wlr_output_layout_output *l_output = data;
+	output_cursor_remove(output_cursor);
+}
+
+static void layout_add(struct wlr_cursor_state *state,
+		struct wlr_output_layout_output *l_output) {
+	struct wlr_cursor_output_cursor *output_cursor =
+		calloc(1, sizeof(struct wlr_cursor_output_cursor));
+	if (output_cursor == NULL) {
+		wlr_log(L_ERROR, "Failed to allocate wlr_cursor_output_cursor");
+		return;
+	}
+	output_cursor->cursor = state->cursor;
+
+	output_cursor->output_cursor = wlr_output_cursor_create(l_output->output);
+	if (output_cursor->output_cursor == NULL) {
+		wlr_log(L_ERROR, "Failed to create wlr_output_cursor");
+		free(output_cursor);
+		return;
+	}
+
+	output_cursor->layout_output_destroy.notify = handle_layout_output_destroy;
+	wl_signal_add(&l_output->events.destroy,
+		&output_cursor->layout_output_destroy);
+
+	wl_list_insert(&state->cursor->state->output_cursors, &output_cursor->link);
```

Comment:
Does it get removed from teh list when `output_cursor` is destroyed? Or does that only happen when the actual output is destroyed?

Reply:
That only happens when the output is destroyed, since there's no `destroy` event in `wlr_output_cursor`. In other words, `wlr_output_cursor`s are destroyed only when the corresponding output is destroyed (or when `wlr_output_cursor_destroy` is called, not relevant here).